src/simplewallet/simplewallet.cpp

```diff
@@ -442,6 +442,21 @@ namespace
       fail_msg_writer() << tr("unexpected error: ") << e.what();
     }
   }
+
+  bool check_file_overwrite(const std::string &filename)
+  {
+    boost::system::error_code errcode;
+    if (boost::filesystem::exists(filename, errcode))
+    {
+      if (boost::ends_with(filename, ".keys"))
+      {
+        fail_msg_writer() << boost::format(tr("File %s likely stores wallet private keys! Use a different file name.")) % filename;
+        return false;
+      }
```

Comment:
Well, I did comment above https://github.com/monero-project/monero/pull/3118#discussion_r161474540. If you're against it, I can drop it.

Reply:
Ah, sorry. I don't really like singling out things, because then why not single other files ? It's not a very strong dislike though, it just feels wrong to me.

Reply:
I'd like to emphasize that this patch was originally motivated by uneducated users' accidental overwrite of .keys files, which might mean the complete loss of their funds if they didn't keep the wallet seed elsewhere. I think it's reasonable to single out .keys files because they are irrecoverable once overwritten, unlike other files (e.g. outputs, key images, multisig info, etc).

It'd be nice if more people could voice their opinions.
CC: @PsychicCat